Doc/library/urllib.request.rst

```diff
@@ -1557,9 +1549,38 @@ some point in the future.
    :synopsis: Response classes used by urllib.
 
 The :mod:`urllib.response` module defines functions and classes which define a
-minimal file like interface, including ``read()`` and ``readline()``. The
-typical response object is an addinfourl instance, which defines an ``info()``
-method and that returns headers and a ``geturl()`` method that returns the url.
-Functions defined by this module are used internally by the
-:mod:`urllib.request` module.
+minimal file-like interface, including ``read()`` and ``readline()``.
+Functions defined by this module are used internally by the :mod:`urllib.request` module.
+The typical response object is a :class:`urllib.response.addinfourl` instance:
+
+.. class:: urllib.response.addinfourl
```

Comment:
I think this is an internal class that was written to enhance other classes, but should not be used or documented on its own.  What do you think?

Reply:
An `addinfourl` object is returned from `urllib.request.urlopen()` though, so I think it needs to be documented in this case.

Reply:
Ah, looking at the existing doc in the paragraph above, I agree!

Reply:
Just wish the class had been named after its role (Response) rather than what it does.